### ISSUE TITLE
Support for Kyma default gateway in API Gateway module

### ISSUE THREAD

**Issue by strekm** [kyma-project/api-gateway#532]
**Description**

APIGateway CR has root property called `enableKymaGateway`, which is by default set to true. There are 2 scenarios that needs to be handled: garden and rest. 
On gardener cluster gardener domain be should read from shoot info and used for setting up Istio Gateway CR. Gardener Certificate CR should be created and generated secret should be referenced in Gateway CR. DNSEntry CR should be created with IP / host from Istio ingressgateway SVC. 
On other clusters static certificate should be used to create a secret referenced in Gateway CR. Static local domain should be used for setting up Gateway CR.

Option to provide configuration by user is not supported anymore. In the future user will be able to provide own default domain and create own gateways (out of the scope of this task).

When `enableKymaGateway` is set to false or completely removed, all related resources mentioned above should be deleted. This should be guarded and if any APIRule is present APIGateway CR should be set to error. In the future this logic should be improved.

ACs:
- [x] default gateway set up properly on gardener and other clusters
  - [x] Add Gardener DNSEntry
  - [x] Add Gardener Certificate
  - [x] Add static Certificate in Secret for non-Gardener
  - [x] Suppot for IP-based and DNS-based load balancer services
- [x] gateway removed is there are no APIRules present and `enableKymaGateway` is set to false or removed
- [x] Kyma gateway removal is blocked by existing APIRules and the status is updated to error (discuss, since warning might be better)
- [x] integration test for k3d and gardener
- [x] documentation 

PR: 
- https://github.com/kyma-project/api-gateway/pull/537
- https://github.com/kyma-project/api-gateway/pull/582

**Reasons**

modularisation

**DoD:**
- [x] provide documentation
~~- [ ] release notes and What's New updates for Kyma customers~~
- [x] provide unit tests
- [x] provide integration tests
- [x] test on production-like environment
~~- [ ] verify resource limits~~
~~- [ ] followup issue~~
~~- [ ] create release and bump in kyma~~
- [x] PR reviewer will verify code coverage and evaluate if it is acceptable

**Attachments**
part of: https://github.com/kyma-project/api-gateway/issues/130
https://github.com/kyma-project/kyma/tree/main/resources/certificates
https://github.com/kyma-project/kyma/blob/main/resources/istio-resources/templates/gateway.yaml

**Comment by triffer** [comment]
There are some open questions:
- Should we Set the APIGateway CR status to `Error` if we cannot disable the Kyma Gateway, because APIRules exist? I think `Warning` would be better.
- Should we also implement the blocking of the deletion of APIGateway CR by existing APIRules in this issue?
- The Kyma Gateway will now be owned by API Gateway operator and Istio operator. That means disabling it in API Gateway operator will not remove it, since Istio Operator will bring it back at some point. On the next default reconciliation API Gateway Operator will remove it again if it exists.
